package.json

```diff
@@ -18,14 +18,16 @@
     "src/",
     "index.js"
   ],
+  "types": "types",
   "scripts": {
     "test": "mocha --timeout 10000",
     "lint": "npm run lint:js",
     "lint:fix": "npm run lint:fix:js",
     "lint:js": "npm run lint:js:eslint && npm run lint:js:types",
     "lint:fix:js": "eslint --fix .",
     "lint:js:eslint": "eslint .",
-    "lint:js:types": "!(npx tsc --allowJs --noEmit $(cat jsconfig.json | jq -r '.compilerOptions | to_entries | map([\"--\\(.key)\",.value]) | flatten | join(\" \")') src/**.js bin/**.js examples/**.js | grep \"^\\(src\\|bin\\|test\\|examples\\)/\") # comment any other passed arguments"
+    "lint:js:types": "!(npx tsc --allowJs --noEmit $(cat jsconfig.json | jq -r '.compilerOptions | to_entries | map([\"--\\(.key)\",.value]) | flatten | join(\" \")') src/**.js bin/**.js examples/**.js | grep \"^\\(src\\|bin\\|test\\|examples\\)/\") # comment any other passed arguments",
```

Comment:
Just to confirm here---ultimately this did end up working?

Reply:
No, I spent several hours trying to get it to work through other methods that don't involve having to parse the .json file directly (would solve the problem on CI while simplifying the command heavily) but I couldn't get it to work, so it's still broken on CI runners.

Probably the easiest way to solve that problem would be patching/upgrading the version of `jq` used on CI, as that is what causes the breakage. If you feel that's a good solution I'll bundle it into this PR.

Reply:
Let's actually pull it to a new PR if that's okay---this one's getting big and starting to drift from coherence haha.

Reply:
> I spent several hours trying to get it to work through other methods that don't involve having to parse the .json file directly

Side note: I did this when I originally landed on this solution too haha. It's brutal.

Reply:
> Let's actually pull it to a new PR if that's okay

On it!

> Side note: I did this when I originally landed on this solution too haha. It's brutal.

Yeah, I spent way too much time trying to get it to work with tsc's `--project`, which seems to be made for this type of use-cases, but while the options were correctly parsed I kept running into problems :confounded: